File: spring-springmvc/src/main/java/com/bh/controller/UserController.java
package com.bh.controller;


import com.bh.entity.User;
import com.bh.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
@RequestMapping("/user")
public class UserController {

    @Autowired
    private UserService userService;

    @RequestMapping("/register")
    public String register(User user) {
        userService.register(user);
        return "regOk";
    }

}

File: kotlin/kotlin-3/src/main/java/com/bh/kotlin/controller/UserController.kt
package com.bh.kotlin.controller

import com.bh.kotlin.model.UserInfo
import com.bh.kotlin.service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/user")
class UserController @Autowired constructor(private val userService: UserService) {

    @GetMapping("getUserInfo")
    fun getUserInfo(): ResponseEntity<UserInfo> {
        return userService.getUserInfo()
    }
}